### PR TITLE
Fix: use of faker gem in ingredient tests

### PR DESCRIPTION
**Pull request by moelheni**

### CODE REVIEW

**Comment by moelheni** [comment]
@SamiDghim  @thameursaadi please do a review for this code.

File: spec/controllers/ingredient_recipe_controller_spec.rb [hunk 1]
```diff
@@ -3,78 +3,62 @@
...
+      expect(response).to have_http_status(:success)
       expect(IngredientRecipe.find_by_quantity(1)).to_not be_nil
     end
 
     it "returns error when recipe missing" do
-      ing = Ingredient.create!(:name => "Ing")
-      post :new, format: :json,:recipe_id => nil,:ingredient_id => ing.id,:quantity => 1,:unit => "g"
-       expect(response).to have_http_status(500)
+      post :new, format: :json,:recipe_id => nil,:ingredient_id => ingredient.id,:quantity => 1,:unit => "g"
+      expect(response).to have_http_status(500)
```
Comment (SamiDghim) [inline spec/controllers/ingredient_recipe_controller_spec.rb]: be more precise in your test , 500 is server error , 
you can use status code 422 in this case

Comment (moelheni) [inline spec/controllers/ingredient_recipe_controller_spec.rb]: I removed this kind of tests.
Model already does the job

**Comment by moelheni** [comment]
@achr3f  @SamiDghim can I merge? test it

File: app/controllers/ingredient_controller.rb [hunk 2]
```diff
@@ -1,15 +1,6 @@
...
-      ingredient = Ingredient.create(name: params[:name])
-      if ingredient.valid?
-        
-      else
-        render status: 500
-      end
-    else
-      render status: 500
-    end
+    ingredient = Ingredient.create(name: params[:name])
```
Comment (achr3f) [inline app/controllers/ingredient_controller.rb:3]: unused variable `ingredient`

File: app/controllers/ingredient_recipe_controller.rb [hunk 3]
```diff
@@ -1,44 +1,22 @@
 class IngredientRecipeController < ApplicationController
   def new
-  	if params[:recipe_id] and params[:ingredient_id] and params[:quantity] and params[:unit]
-  		ir = IngredientRecipe.create(recipe_id: params[:recipe_id],
+  	ir = IngredientRecipe.create(recipe_id: params[:recipe_id],
```
Comment (achr3f) [inline app/controllers/ingredient_recipe_controller.rb]: same unused `ir` variable

File: app/controllers/ingredient_recipe_controller.rb [hunk 4]
```diff
@@ -1,44 +1,22 @@
...
                                    ingredient_id: params[:ingredient_id],
   			                           quantity: params[:quantity],
```
Comment (achr3f) [inline app/controllers/ingredient_recipe_controller.rb]: indent

File: app/controllers/ingredient_recipe_controller.rb [hunk 5]
```diff
@@ -1,44 +1,22 @@
...
-  	end
   end
 
   def index
-    if params[:recipe_id] 
-  	  @ingredient_recipes = Recipe.find(params[:recipe_id]).ingredient_recipes
-    else
-      render :status => 500
-    end  
+  	@ingredient_recipes = Recipe.find(params[:recipe_id]).ingredient_recipes
```
Comment (achr3f) [inline app/controllers/ingredient_recipe_controller.rb]: 4 spaces ??

File: spec/controllers/ingredient_controller_spec.rb [hunk 6]
```diff
@@ -6,92 +6,43 @@
...
   end
 
   describe "GET #delete" do
+    let!(:ingredient) { create(:ingredient,:with_name) }
 
     it "deletes when a record is found" do
-      i = Ingredient.new(:name => "Ing")
-      i.save
-      json = { :format => 'json', :id => i.id }
+      json = { :format => 'json', :id => ingredient.id }
```
Comment (achr3f) [inline spec/controllers/ingredient_controller_spec.rb:41]: ruby 1.9 syntax for hash

**Comment by achr3f** [comment]
FIX AND SHIP IT

File: app/controllers/ingredient_controller.rb (hunk 2 above)
Comment (moelheni) [inline app/controllers/ingredient_controller.rb:3]: already changed, i don't know why you see the old version, refresh or sth

**Pull request merged**